tools/curl/test/test_trace_parser.py

```diff
@@ -46,6 +46,10 @@ def test_google():
     assert res["response"]["headers"]["Accept-Ranges"] == "none"
     assert res["response"]["headers"]["Vary"] == "Accept-Encoding"
     assert res["response"]["headers"]["Transfer-Encoding"] == "chunked"
+    try:
+        json.dumps(res)
+    except Exception as e:
+        assert False, e
```

Comment:
Suspicious...

If your goal is to know if json is correct I suggest you decode the JSON as the first step and check the previous headers against the decoded object.

Reply:
It's about the extrange characters in body. I'm fine if content goes out correctly.
Remember the extrage base64 encoding approach upwards?

Reply:
But why the previous header checkings then? Maybe it should be split in two tests?

Reply:
yes, that's right two test will be great, doing